### ISSUE TITLE
Assistance with Tcpreplay bad addr/len

### ISSUE THREAD

**Issue by fklassen** [luigirizzo/netmap#582]

**Comment by vmaffione** [comment]
You need to increase `/sys/module/netmap/parameters/buf_size`.

**Comment by giuseppelettieri** [comment]
Yes, the defauit buffer size is 2048. You can either increase that (no change needed in the app, but possibly wasteful) or support ```NS_MOREFRAG``` in ```tcpreplay```. Basically, you need to use several consecutive slots for a single message and set ```NS_MOREFRAG``` in the slot flags of all but the last slot.

**Comment by fklassen** [comment]
I am having Tcpreplay crashes when I increase the buffer size dynamically. Does the value have to be a power of two? I am attempting to accept 9018 frames. I'm not sure if I have to alter Tcpreplay source code to make this work.

**Comment by fklassen** [comment]
> or support `NS_MOREFRAG` in `tcpreplay`. Basically, you need to use several consecutive slots for a single message and set `NS_MOREFRAG` in the slot flags of all b

I'll add a ticket to add that flag. Question: Does the flag work dynamicallay, e.g. if a 1500 byte packet comes in, will it only take one 2048 byte buffer?

**Comment by giuseppelettieri** [comment]
The buffer size can be any value between 64 and 65536. it doesn't need to be a power of 2, but the number you enter may rounded to a multiple of a cacheline (you should see a message in the logs). The new value is not used while there are netmap applications still running.

**Comment by giuseppelettieri** [comment]
> Does the flag work dynamicallay, e.g. if a 1500 byte packet comes in, will it only take one 2048 byte buffer?
it is a per-slot flag. If you have a packet that fits a single slot, you simply don't set the flag. If you have a packet that needs 4 slots, you set the flag in the first three.

**Comment by fklassen** [comment]
> it is a per-slot flag. If you have a packet that fits a single slot, you simply don't set the flag. If you have a packet that needs 4 slots, you set the flag in the first three.

That makes sense. Thanks for clarifying. Opened [appneta/tcpreplay:#534](https://github.com/appneta/tcpreplay/issues/534) to track this issue.
